### PR TITLE
Fix conditional formatting render with black background when values >= 1,000

### LINKED ISSUE #13892: Pulses with Conditional Formatting is displayed with black background, when value is above 999
**Describe the bug**
When using Conditional Formatting on a table, then column values above 999 (with thousand separator) will show a black background in Pulse.

**To Reproduce**
1. Simple question > Sample Dataset > Orders
2. Summarize Count by People.Source and Product.Category
3. Change visualization to Table and un-pivot in Settings and set Conditional Formatting on Count with style "Color range"
image
4. Save question and try creating a Pulse, which then shows all values above 999 with black background
image

**Information about your Metabase Installation:**
Tested 0.34.3 thru 0.37.2

**Additional context**
Related to #3735

The backend invokes JS to get a cell color, however, it passes not javascript numbers but objects `toString` of which returns formatted numbers with the following keys:
```
num_str,__meta,__extmap,hasheq,withMeta,meta,assoc,valAt,entryAt,without,getLookupThunk,equiv,remove,get,put,equals,toString,values,hashCode,clear,isEmpty,size,iterator,count,entrySet,putAll,empty,cons,containsKey,keySet,containsValue,seq,to_str,assocEx,spliterator,forEach,replace,replaceAll,merge,putIfAbsent,compute,computeIfAbsent,getOrDefault,computeIfPresent
```
Then d3.scale code checks if the provided value is a number. In case with `999` string gets converted to a number and the check passes but `1,000` string is `NaN`. To fix that, BE should provide numbers instead of objects.  

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

### PR DESCRIPTION
**Pull request by qnkhuat**
Fixes #13892 and #28725

When rendering a table for pulse, BE calls a js function to get the appropriate background color for each cell.
We had this bug because when calling the js function, we provided the js function with an object `(NumericWrapper. "1,000")` instead of a number.

the js function is nice enough to detect this is not a number and automatically call `.toString` on it, but since we formatted the number with a comma ("1,000") it doesn't know how to convert this to a number and return a default black color.

The fix is to make sure we unwrap value and parse the number correctly.

### CODE REVIEW

File: src/metabase/pulse/render/color.clj [hunk 1]
```diff
@@ -49,9 +54,23 @@
...
   ^String [color-selector cell-value column-name row-index]
-  (.asString (js/execute-fn color-selector cell-value row-index column-name)))
+  (let [cell-value (cond
+                     (instance? NumericWrapper cell-value)
+                     (-> ^NumericWrapper cell-value
+                         .toString
+                         (str/replace #"," "")
+                         parse-number)
+                     :else
+                     cell-value)]
```
Comment (metamben) [inline src/metabase/pulse/render/color.clj:75]:
```suggestion
  (let [cell-value (if (instance? NumericWrapper cell-value)
                     (-> ^NumericWrapper cell-value
                         .toString
                         (str/replace #"," "")
                         parse-number)
                     cell-value)]
```

File: src/metabase/pulse/render/color.clj [hunk 2]
```diff
@@ -49,9 +54,23 @@
...
+   (.asString (js/execute-fn color-selector cell-value row-index column-name))))
```
Comment (metamben) [inline src/metabase/pulse/render/color.clj:76]: The indentation seems to be wrong.

File: test/metabase/pulse/render/test_util.clj [hunk 3]
```diff
@@ -37,6 +37,7 @@
                                :highlight_row true}
                               {:columns       ["c"]
                                :type          "range"
+                               :operator      "="
```
Comment (metamben) [inline test/metabase/pulse/render/test_util.clj:40]: How is this related to the change?

**Review by metamben — APPROVED** [review]
LGTM

File: src/metabase/pulse/render/color.clj [hunk 4]
```diff
@@ -49,9 +54,23 @@
...
   "Get the correct color for a cell in a pulse table. Returns color as string suitable for use CSS, e.g. a hex string or
   `rgba()` string. This is intended to be invoked on each cell of every row in the table. See `make-color-selector`
   for more info."
   ^String [color-selector cell-value column-name row-index]
-  (.asString (js/execute-fn color-selector cell-value row-index column-name)))
+  (let [cell-value (cond
+                     (instance? NumericWrapper cell-value)
+                     (-> ^NumericWrapper cell-value
+                         .toString
+                         (str/replace #"," "")
```
Comment (tsmacdonald) [inline src/metabase/pulse/render/color.clj:72]: Will this fail in, e.g., Spanish, where `999 + 1 + ½` is written `1.000,5` instead of `1,000.5`?

**Review by tsmacdonald — CHANGES_REQUESTED** [review]
worried about localization, but not quite sure if `NumericWrapper` uses an arbitrary locale (though if it doesn't, why do we localize at all?)

File: test/metabase/pulse/render/test_util.clj (hunk 3 above)
Comment (qnkhuat) [reply test/metabase/pulse/render/test_util.clj:40]: it's unrelated, I just add it for the sake of completeness. I believe `=` was the default operator and we could omit it.

**Review by tsmacdonald — APPROVED** [review]
:boat:

**Pull request merged**